manim/utils/hashing.py

```diff
@@ -30,56 +32,127 @@ def default(self, obj):
+        """Check for circular reference at each iterable that will go through the JSONEncoder, as well as key of the wrong format.
 
-        if isinstance(obj, dict):
-            return {key_to_hash(k): self._encode_dict(v) for k, v in obj.items()}
-        return obj
+        If a key with a bad format is found (i.e not a int, string, or float), it gets replaced byt its hash using the same process implemented here.
+        If a circular reference is found within the iterable, it will be replaced by the string "already processed".
+
+        Parameters:
+        -----------
+        iterable : :class:`Iterable`
+            The iterable to check.
+        """
+
+        def _key_to_hash(key):
+            return zlib.crc32(json.dumps(key, cls=CustomEncoder).encode())
+
+        def _iter_check_list(lst):
+            # We have to make a copy, as we don't want to touch to the original list
+            # A deepcopy isn't necessary as it is already recursive.
+            lst_copy = copy.copy(lst)
+            for i, el in enumerate(lst):
+                if not isinstance(lst, tuple):
+                    lst_copy[i] = self._handle_already_processed(
+                        el
+                    )  # ISSUE here, because of copy.
+                if isinstance(el, (list, tuple)):
+                    lst_copy[i] = _iter_check_list(el)
+                elif isinstance(el, dict):
+                    lst_copy[i] = _iter_check_dict(el)
+            return lst_copy
+
+        def _iter_check_dict(dct):
+            # We have to make a copy, as we don't want to touch to the original dict
+            # A deepcopy isn't necessary as it is already recursive.
+            dct_copy = copy.copy(dct)
+            for k, v in dct.items():
+                dct_copy[k] = self._handle_already_processed(v)
+                # We check if the k is of the right format (supporter by Json)
+                if not isinstance(k, (str, int, float, bool)) and k is not None:
+                    k_new = _key_to_hash(k)
+                    # We delete the value coupled with the old key, as the value is now coupled with the new key.
+                    dct_copy[k_new] = dct_copy[k]
+                    del dct_copy[k]
+                else:
+                    k_new = k
+                if isinstance(v, dict):
+                    dct_copy[k_new] = _iter_check_dict(v)
+                elif isinstance(v, (list, tuple)):
+                    dct_copy[k_new] = _iter_check_list(v)
+            return dct_copy
+
+        if isinstance(iterable, (list, tuple)):
+            return _iter_check_list(iterable)
+        elif isinstance(iterable, dict):
+            return _iter_check_dict(iterable)
+        raise Exception("wtf")
 
     def encode(self, obj):
-        return super().encode(self._encode_dict(obj))
+        """Overwriting of JSONEncoder.encode, to make our own process.
+
+        Parameters:
+        -----------
+        obj: :class:Any
```

Comment:
There is no class named Any, also forgot backticks, but we're not gonna need them anyway

```suggestion
        obj: Any
```